File: BusInfo/app/src/main/java/com/daeseong/businfo/BusApplication.java
package com.daeseong.businfo;

import android.app.Application;
import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.res.Configuration;
import android.graphics.Color;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.util.Log;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

public class BusApplication extends Application {

    private static final String TAG = BusApplication.class.getSimpleName();

    public String API_Key ="%2FSWbuoncrZtSM3DaBUA4PJVxqJMFKs0Eu%2F%2FzgFQf8dvVjzIi8ESOjmRaQtAkLKoQUS3S%2BZy%2FwLwR08%2BCT9BWuA%3D%3D";

    private static BusApplication mInstance;
    public static synchronized BusApplication getInstance() {
        return mInstance;
    }

    private static Context mContext;
    public static Context getAppContext(){
        return mContext;
    }

    private static Toast toast;
    public static void showToast(String sMsg, boolean isLengthLong) {

        LayoutInflater inflater = (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View view = inflater.inflate(R.layout.toast_layout, null);

        TextView tvToast = view.findViewById(R.id.tvtoast);
        tvToast.setMaxLines(1);
        tvToast.setTextColor(Color.parseColor("#000000"));
        tvToast.setText(sMsg);

        toast = new Toast(mContext);
        toast.setGravity(Gravity.BOTTOM, 0, 0);
        toast.setDuration(isLengthLong ? Toast.LENGTH_LONG : Toast.LENGTH_SHORT);
        toast.setView(view);
        toast.show();
    }

    public static void cancelToast() {
        try {
            if (toast != null) {
                toast.cancel();
            }
        } catch (Exception e) {
        }
    }

    private ClipboardManager clipboardManager;
    private ClipData clipData;

    public void setClipboardText(String input) {
        try {
            if (clipboardManager != null) {
                clipData = ClipData.newPlainText("text", input);
                clipboardManager.setPrimaryClip(clipData);
            }
        } catch (Exception ex) {
        }
    }

    public String getClipboardText() {
        String data = "";
        try {
            if (clipboardManager != null && clipboardManager.hasPrimaryClip()) {
                clipData = clipboardManager.getPrimaryClip();
                ClipData.Item item = clipData.getItemAt(0);
                data = item.getText().toString();

                if (!String_util.isNumeric(data)) {
                    data = "";
                }
            }
        } catch (Exception ex) {
        }
        return data;
    }

    public void clearClipboardText() {
        try {
            if (clipboardManager != null) {
                ClipData data = clipData.newPlainText("", "");
                clipboardManager.setPrimaryClip(data);
            }
        } catch (Exception ex) {
        }
    }

    public static boolean isNetworkAvailable(Context context) {
        ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
        if (networkInfo != null && networkInfo.isConnected()){
            return true;
        }
        return false;
    }

    @Override
    public void onCreate() {
        super.onCreate();

        mInstance = this;
        mContext = getApplicationContext();

        try {
            clipboardManager = (ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
        } catch (Exception e) {
        }
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
    }
}

File: BusInfo(kotlin)/app/src/main/java/com/daeseong/businfo/BusApplication.kt
package com.daeseong.businfo

import android.app.Application
import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.res.Configuration
import android.graphics.Color
import android.net.ConnectivityManager
import android.view.Gravity
import android.view.LayoutInflater
import android.widget.TextView
import android.widget.Toast

class BusApplication : Application() {

    companion object {

        private lateinit var mInstance: BusApplication
        private lateinit var mContext: Context
        private var toast: Toast? = null

        private lateinit var clipboardManager: ClipboardManager
        private lateinit var clipData: ClipData

        @JvmStatic
        fun getInstance(): BusApplication {
            return mInstance
        }
    }

    val API_Key = "%2FSWbuoncrZtSM3DaBUA4PJVxqJMFKs0Eu%2F%2FzgFQf8dvVjzIi8ESOjmRaQtAkLKoQUS3S%2BZy%2FwLwR08%2BCT9BWuA%3D%3D"

    override fun onCreate() {
        super.onCreate()

        mInstance = this
        mContext = applicationContext

        try {
            clipboardManager = getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        } catch (e: Exception) {
        }
    }

    fun showToast(sMsg: String, isLengthLong: Boolean) {
        val inflater = mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        var view = inflater.inflate(R.layout.toast_layout, null)

        val tvToast = view.findViewById<TextView>(R.id.tvtoast)
        tvToast.maxLines = 1
        tvToast.setTextColor(Color.parseColor("#000000"))
        tvToast.text = sMsg

        toast = Toast(mContext)
        toast?.apply {
            setGravity(Gravity.BOTTOM, 0, 0)
            duration = if (isLengthLong) Toast.LENGTH_LONG else Toast.LENGTH_SHORT
            view = view
            show()
        }
    }

    fun cancelToast() {
        try {
            toast?.cancel()
        } catch (e: Exception) {
        }
    }

    fun setClipboardText(input: String) {
        try {
            clipData = ClipData.newPlainText("text", input)
            clipboardManager.setPrimaryClip(clipData)
        } catch (ex: Exception) {
        }
    }

    fun getClipboardText(): String {
        var data = ""
        try {
            if (clipboardManager.hasPrimaryClip()) {
                clipData = clipboardManager.primaryClip!!
                val item = clipData.getItemAt(0)
                data = item.text.toString()

                if (!String_util.isNumeric(data)) {
                    data = ""
                }
            }
        } catch (ex: Exception) {
        }
        return data
    }

    fun clearClipboardText() {
        try {
            clipData = ClipData.newPlainText("", "")
            clipboardManager.setPrimaryClip(clipData)
        } catch (ex: Exception) {
        }
    }

    fun isNetworkAvailable(context: Context): Boolean {
        val connectivityManager = context.getSystemService(CONNECTIVITY_SERVICE) as ConnectivityManager
        val networkInfo = connectivityManager.activeNetworkInfo
        return networkInfo != null && networkInfo.isConnected
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
    }
}
